mlflow/legacy_databricks_cli/configure/provider.py

```diff
@@ -265,8 +272,12 @@ def get_config(self):
         refresh_token = os.environ.get("DATABRICKS_REFRESH_TOKEN")
         insecure = os.environ.get("DATABRICKS_INSECURE")
         jobs_api_version = os.environ.get("DATABRICKS_JOBS_API_VERSION")
+        client_id = os.environ.get("DATABRICKS_CLIENT_ID")
```

Comment:
Can we verify that these are set correctly within Jobs?

Reply:
I ran testing code in PR description in Databricks serverless and Databricks jobs notebook, all of them work well.

But note that if we run MLflow in notebook of databricks runtime (either interactive notebook or a job), MLflow by default reads credentials from the **dynamic token" generated by `dbutils` (see `_dynamic_token_config_provider` in my code)

Do we have some use case that runs in databricks runtime but not in a notebook ? I can test the case as well, in such case, `dbutils` might not be available, then dynamic token is not available, MLflow will fallback to other ways of reading creds such as from Profile config file or environmental variable.

Reply:
Executing a .whl, perhaps? I'm not sure if run submit is still a viable option or not as well.

Reply:
confirmed offline. addressed.